### PR TITLE
feat(pci.ai): add manual data sync in notebook and apps

### PR DESCRIPTION
**Pull request by abullet33**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #datatr-162 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description
Adding possibility to launch manual data synchronization in AI apps and notebook

## Translation 🎈
- [x] CMT-6731

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/attach-data/attach-data.controller.js [hunk 1]
```diff
@@ -34,4 +34,19 @@ export default class {
   refreshMessages() {
     this.messages = this.messageHandler.getMessages();
   }
+
+  manualDataSync(volume) {
+    if (volume) {
+      const selectedVolume = this.app.status.volumes.find(
+        (vol) => vol.mountPath === volume.mountPath,
+      );
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/attach-data/attach-data.controller.js:42]: **niptpick:** Avoid short names
```suggestion
      const selectedVolume = this.app.status.volumes.find(
        ({ mountPath }) => mountPath === volume.mountPath,
      );
```

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js [hunk 2]
```diff
@@ -33,4 +33,19 @@ export default class {
...
+        (vol) => vol.mountPath === volume.mountPath,
+      );
+      this.goToManualDataSync(
+        this.notebook.id,
+        selectedVolume.id,
+        selectedVolume.mountPath,
+      );
+    } else {
+      this.goToManualDataSync(this.notebook.id);
+    }
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js:49]: Same as https://github.com/ovh/manager/pull/10410#discussion_r1411905522

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js [hunk 3]
```diff
@@ -33,4 +33,19 @@ export default class {
   refreshMessages() {
     this.messages = this.messageHandler.getMessages();
   }
+
+  manualDataSync(volume) {
+    if (volume) {
+      const selectedVolume = this.notebook.status.volumes.find(
+        (vol) => vol.mountPath === volume.mountPath,
+      );
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js:41]: Same as https://github.com/ovh/manager/pull/10410#discussion_r1411906924

**Review by anooparveti — CHANGES_REQUESTED** [review]
**question:** Cant we just define dataSync component once and reuse it both in AI-Apps and AI-Notebooks?

**Review by qpavy — DISMISSED** [review]

**Comment by abullet33** [comment]
> **question:** Cant we just define dataSync component once and reuse it both in AI-Apps and AI-Notebooks?

Hey @anooparveti, 
Yep you are right. It will also be used for ai training so I create a new component that will be used in all AI Product. Thanks for the tips :) 
Regards

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            <select
+                name="type"
+                id="type"
+                class="oui-select__input"
+                data-ng-model="$ctrl.currentDataSyncType"
+                required
+            >
+                <option ng-repeat="type in $ctrl.dataSyncType" ng-value="type"
+                    >{{type}}</option
+                >>
```
Comment (qpavy) [inline packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html:24]:
```suggestion
                >
```
Typo ?

**Review by qpavy — DISMISSED** [review]

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+angular
+  .module(moduleName, [
+    'oui',
+    'ngTranslateAsyncLoader',
+    'pascalprecht.translate',
+  ])
+  .component('ovhManagerPciProjectAiDataSync', component)
+  .service('PciProjectAppService', appService)
+  .service('PciProjectNotebookService', notebookService)
+  .service('PciProjectJobService', jobService)
```
Comment (jperchoc) [inline packages/manager/modules/pci/src/components/pci-ai-data-sync/index.js:23]: It seems that those services are not used, I think you can remove them

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        <label class="oui-select">
+            <select
+                name="type"
+                id="type"
+                class="oui-select__input"
+                data-ng-model="$ctrl.currentDataSyncType"
+                required
+            >
+                <option ng-repeat="type in $ctrl.dataSyncType" ng-value="type"
+                    >{{type}}</option
```
Comment (jperchoc) [inline packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html:23]:
```suggestion
                <option data-ng-repeat="type in $ctrl.dataSyncType" data-ng-value="type" data-ng-bind="type"></option
```

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/index.js (hunk 5 above)
Comment (abullet33) [reply packages/manager/modules/pci/src/components/pci-ai-data-sync/index.js]: fixed :)

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html (hunk 6 above)
Comment (abullet33) [reply packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.html]: fixed :)

**Review by qpavy — APPROVED** [review]

File: packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.controller.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+export default class PciAiDataSyncCtrl {
+  /* @ngInject */
+  constructor($translate, atInternet) {
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+  }
+
+  $onInit() {
+    this.dataSyncType = DATA_SYNC_TYPES;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/components/pci-ai-data-sync/pci-ai-data-sync.controller.js:11]: **suggestion:** To indicate that the variable is a constant.
```suggestion
    this.DATA_SYNC_TYPES = DATA_SYNC_TYPES;
```

**Review by anooparveti — DISMISSED** [review]

**Review by jperchoc — DISMISSED** [review]

**Review by qpavy — APPROVED** [review]

**Review by antonymarion — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**